relayer/src/link/operational_data.rs

```diff
@@ -178,6 +178,19 @@ impl OperationalData {
 
         Ok(tm)
     }
+
+    pub fn has_packet_msgs(&self) -> bool {
+        self.batch.iter().any(|msg| {
+            matches!(
+                msg.event,
+                IbcEvent::ReceivePacket(_)
+                    | IbcEvent::WriteAcknowledgement(_)
+                    | IbcEvent::AcknowledgePacket(_)
+                    | IbcEvent::TimeoutPacket(_)
+                    | IbcEvent::TimeoutOnClosePacket(_)
+            )
+        })
```

Comment:
There's a `IbcEvent::packet(&self) -> Option<&Packet>` function that we could reuse here -> https://github.com/informalsystems/ibc-rs/blob/master/modules/src/events.rs#L446
```rust
pub fn has_packet_msgs(&self) -> bool {
    self.batch.iter().any(|msg| msg.event.packet().is_some())
}
```
But I'm a little confused if we should because I don't think we should take `SendPacket` events into account here. 🤔

Reply:
I think I was mistaken about not checking for `SendPacket` events. Just updated the code to use the `msg.event.packet().is_some()` I described above, although I am still not completely sure if this is full-proof or if we need to match by `msg.msg.type_url` to be certain if we're dealing with packet related messages.